src/common.h

```diff
@@ -93,7 +93,7 @@ typedef signed long long int64;
 #define TRACK_NUM(a,c)	m->mod.xxp[a]->index[c]
 #define EVENT(a,c,r)	m->mod.xxt[TRACK_NUM((a),(c))]->event[r]
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW32__)
```

Comment:
@sezero, this line fixes the compile failure in debug mode when using MinGW: the debug.c takes the `D_` macro and fails because of the syntax mess caused by the wrong macro.  Maybe just put the `_WIN32` check instead?

Reply:
If you want, you may commit this to the main branch separately. Note, this thing fixes the debug build, at release builds there is no fault because all code of `src/win32/debug.c` got disabled

Reply:
Looking at it

Reply:
I configured using autotools, manually added `-DDEBUG` to CFLAGS, and got no issues.
Can you show the errors you are getting?

Reply:
> I configured using autotools, manually added -DDEBUG to CFLAGS, and got no issues.
Can you show the errors you are getting?

this fails on Windows only, as it happens ant Windows-only code, gonna try some, I'll need to boot my VM again...

Reply:
Please also define the `-D_DEBUG` in addition

Reply:
Failed with:
```
CFLAGS="-DDEBUG -D_DEBUG" ./configure
```
And result:
```
In file included from src/win32/debug.c:7:0:
src/common.h:150:21: error: expected identifier or '(' before 'do'
 #define D_(args...) do { \
                     ^
src/win32/debug.c:9:6: note: in expansion of macro 'D_'
 void D_(const char *format, ...)
      ^~
src/common.h:153:4: error: expected identifier or '(' before 'while'
  } while (0)
    ^
src/win32/debug.c:9:6: note: in expansion of macro 'D_'
 void D_(const char *format, ...)
      ^~
Makefile:83: recipe for target 'src/win32/debug.lo' failed
mingw32-make: *** [src/win32/debug.lo] Error 1
```
![изображение](https://user-images.githubusercontent.com/6751442/125365138-5ff0d280-e37c-11eb-893e-bfe23c27e2ff.png)


Mainly, because the `-D_DEBUG` flag is getting being defined to enable certain debug stuff at WinAPI headers and other cases like asserts, etc.

Reply:
So, I guess, maybe instead of MINGW condition, check the _DEBUG presence? I mean this macro is the source of this issue

Reply:
I mean:
```c
#if defined(_MSC_VER) || defined(_DEBUG)
```

Reply:
OK, found the issue and pushed https://github.com/libxmp/libxmp/commit/46acc04ad48077b7c5abca01d54c45ad65805960

I guess you can safely remove your commit from the patchset